src/eval-frame/reducers/eval-frame-reducer.js

```diff
@@ -79,7 +79,12 @@ const notebookReducer = (state = newNotebook(), action) => {
     case 'UPDATE_VALUE_IN_HISTORY': {
       const history = [...state.history]
       const historyEntry = history.find(h => h.historyId === action.historyId)
-      if (historyEntry) { historyEntry.value = action.value }
+      if (historyEntry) {
```

Comment:
i don't think we really need to add functionality to change historyTypes, see other comments